components/openapi/src/main/scala/pl/touk/nussknacker/openapi/OpenAPIComponentProvider.scala

```diff
@@ -1,69 +1,29 @@
 package pl.touk.nussknacker.openapi
 
-import cats.data.Validated
-import cats.data.Validated.{Invalid, Valid}
-import com.typesafe.config.{Config, ConfigFactory, ConfigRenderOptions, ConfigValueFactory}
+import com.typesafe.config.Config
 import com.typesafe.scalalogging.LazyLogging
-import io.circe.syntax.EncoderOps
-import net.ceedubs.ficus.Ficus._
 import pl.touk.nussknacker.engine.ModelConfig
-import pl.touk.nussknacker.engine.api.CirceUtil
 import pl.touk.nussknacker.engine.api.component.{ComponentDefinition, ComponentProvider, NussknackerVersion}
 import pl.touk.nussknacker.engine.util.config.ConfigEnrichments._
 import pl.touk.nussknacker.openapi.OpenAPIServicesConfig._
-import pl.touk.nussknacker.openapi.discovery.SwaggerOpenApiDefinitionDiscovery
-import pl.touk.nussknacker.openapi.enrichers.{SwaggerEnricherCreator, SwaggerEnrichers}
-import pl.touk.nussknacker.openapi.parser.ServiceParseError
-
-import scala.jdk.CollectionConverters._
-import scala.util.control.NonFatal
+import pl.touk.nussknacker.openapi.discovery.{CachingOpenApiDefinitionDiscovery, SwaggerOpenApiDefinitionDiscovery}
+import pl.touk.nussknacker.openapi.enrichers.OpenAPIEnricherFactory
 
 class OpenAPIComponentProvider extends ComponentProvider with LazyLogging {
 
   override def providerName: String = "openAPI"
 
-  override def resolveConfigForExecution(config: Config): Config = {
-    // we need to load config to resolve url which can be potentially a system env variable
-    val openAPIsConfig = ConfigFactory.load(config).rootAs[OpenAPIServicesConfig]
-    val services =
-      try {
-        SwaggerOpenApiDefinitionDiscovery.discoverOpenAPIServices(openAPIsConfig)
-      } catch {
-        case NonFatal(ex) =>
-          logger.error("OpenAPI service resolution failed. Will be used empty services lists", ex)
-          List.empty
-      }
-    val servicesToUse = services.collect { case Valid(service) =>
-      ConfigFactory.parseString(service.asJson.spaces2).root()
-    }
-    logErrors(services)
-    config.withValue("services", ConfigValueFactory.fromIterable(servicesToUse.asJava))
```

Comment:
This trick allows OpenAPI services to be passed to a Flink job via configuration during deployment. As a result, the deployed Flink job doesn't need to rediscover the OpenAPI services - their definitions are simply recreated from the provided configuration. Perhaps this behavior should be retained as it was?

Reply:
Is not enough to pass chosen swaggerService to implementation method and return enricher in this dynamic component?